cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -430,7 +433,23 @@ template <>
 hash_value_type CUDA_DEVICE_CALLABLE
 SparkMurmurHash3_32<numeric::decimal128>::operator()(numeric::decimal128 const& key) const
 {
-  return this->compute<__int128_t>(key.value());
```

Comment:
Why we have to implement new code? What's wrong with `compute<__int128_t>`?

Reply:
It's unique to how spark handles decimal128 values, the standard compute function hashes the value directly rather than flipping the endianness and shortening the hashed data.